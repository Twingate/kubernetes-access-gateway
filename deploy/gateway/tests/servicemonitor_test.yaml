suite: Service Monitor
templates:
  - servicemonitor.yaml
tests:
  - it: should enable Prometheus Service Monitor
    set:
      metrics:
        serviceMonitor:
          enabled: true
    asserts:
      - matchSnapshot: {}
  - it: should set `interval`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          interval: 60s
    asserts:
      - equal:
          path: spec.endpoints[0].interval
          value: 60s
  - it: should set `scrapeTimeout`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          scrapeTimeout: 60s
    asserts:
      - equal:
          path: spec.endpoints[0].scrapeTimeout
          value: 60s
  - it: should set `relabelings`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          relabelings:
            - foo: bar
    asserts:
      - contains:
          path: spec.endpoints[0].relabelings
          content:
            foo: bar
  - it: should set `metricRelabelings`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          metricRelabelings:
            - foo: bar
    asserts:
      - contains:
          path: spec.endpoints[0].metricRelabelings
          content:
            foo: bar
  - it: should set `honorLabels`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          honorLabels: true
    asserts:
      - equal:
          path: spec.endpoints[0].honorLabels
          value: true
  - it: should set `additionalLabels`
    set:
      metrics:
        serviceMonitor:
          enabled: true
          additionalLabels:
            foo: bar
    asserts:
      - equal:
          path: metadata.labels.foo
          value: bar
