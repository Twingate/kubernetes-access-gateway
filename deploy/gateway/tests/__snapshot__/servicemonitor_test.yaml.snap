should enable Prometheus Service Monitor:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      endpoints:
        - interval: 30s
          targetPort: metrics
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kubernetes-access-gateway
